nutils/sample.py

```diff
@@ -281,8 +201,13 @@ def eval_sparse(self, funcs:types.tuple[function.asarray], arguments:types.froze
         Optional arguments for function evaluation.
     '''
 
-    funcs = self._prepare_funcs_eval(funcs)
-    return self._eval(funcs, arguments)
+    return eval_integrals_sparse(*map(self.__rmatmul__, funcs), **(arguments or {}))
```

Comment:
why not `evaluable.Tuple((func @ self).lower() for func in funcs).simplified.eval()`, wouldn't that be more efficient than to go back and forth via sparse? also then `eval` and `@` do precisely the same

Reply:
We need the back and forth because `Evaluable.assparse`, like `Evaluable.blocks`, eliminates `Inflate`. This was already the route we took before this PR. With `function.Array @ Sample` we create a `function.Array` that most likely will be evaluated as a sparse array and made dense afterwards (e.g. when using `function.Array.eval` or `sample.eval_integrals`).

Reply:
Ok, I'm aware we currently go via sparse but that's fairly recent and I still fear it's a potential regression, and I thought that `_AtSample` would be in a good position to return assembling directly into the dense array, but I get that that gives issues with sparse functions. Anyhow my observation that `sample.eval(f)` and (f @ sample).eval()` take different routes is wrong I see now. All in all it looks fine on second view.